velox/exec/Task.h

```diff
@@ -1159,13 +1178,15 @@ class Task : public std::enable_shared_from_this<Task> {
   // Base spill directory for this task.
   std::string spillDirectory_;
 
-  // Mutex to ensure only the first caller thread of 'getOrCreateSpillDirectory'
-  // creates the directory.
-  mutable std::mutex spillDirCreateMutex_;
+  // Mutex to ensure only the first caller thread of spill/trace directory
+  // creator creates the directory.
+  mutable std::mutex dirCreateMutex_;
 
   // Indicates whether the spill directory has been created.
   std::atomic<bool> spillDirectoryCreated_{false};
 
+  std::string traceDirectory_;
```

Comment:
Move spillDirectory_ right here (after the corresponding lock annotation)

Reply:
moved